src/payment/strategies/klarna/klarna-payment-strategy.ts

```diff
@@ -121,6 +132,35 @@ export default class KlarnaPaymentStrategy implements PaymentStrategy {
             }));
     }
 
+    private _getUpdateSessionData(billingAddress: BillingAddress, shippingAddress?: Address): KlarnaUpdateSessionParams | {} {
+        if (billingAddress.countryCode === 'US') {
+            return {};
+        }
+
+        const data = {} as KlarnaUpdateSessionParams;
+
+        data.billing_address = this._mapToKlarnaAddress(billingAddress, billingAddress.email);
+
+        if (shippingAddress) {
+            data.shipping_address = this._mapToKlarnaAddress(shippingAddress, billingAddress.email);
```

Comment:
Also, since email is optional, why are we using the billing email address to fill the shipping one? shouldn't just be left blank?

Reply:
Klarna's implementation in BigPay is doing the same thing (adding the billing email to the shipping data), so if I don't follow same process Klarna will throw a mismatch error

Reply:
fair enough